### ISSUE TITLE
To confirm some technical questions

### ISSUE THREAD

**Issue by Kyfafyd** [facebookresearch/Detic#49]
Hi, Thanks for your sharing this nice work!
I would like to confirm the following question:
- The `image encoder in CLIP` is not used here, instead by a zeroshot classifier?
- The text embeddings are precomputed by `CLIP text encoder` saved as `metadata/lvis_v1_clip_a+cname.npy` for LVIS val set, so text encoder can be regarded as fixed in training?
- In zeroshot classifier forward process, what `x` and `classifier` represent for?

**Comment by xingyizhou** [comment]
Hi,

  Thank you for your interest in our work.

  1. I didn't understand the question. Feel free to post it in Chinese. In case it helps, we mean we just use the language encoder (6+6 layers transformer) in CLIP to obtain a single feature (for each word) as the last layer. We did not use the ViT/ ResNet image backbone.
  2. Correct.
  3. Which line of code are you referring to? I will guess x is the region feature after the ROIAlign and two linear layers.


Best,
Xingyi

**Comment by Kyfafyd** [comment]
Hi,

Thanks for your response, 1 and 2 have been addressed!
For 3, this line:
https://github.com/facebookresearch/Detic/blob/f633b7869f87a33fe9ee90b824301ef5e98f0fbf/detic/modeling/roi_heads/zero_shot_classifier.py#L69

**Comment by xingyizhou** [comment]
It is the region feature after the ROIAlign and two linear layers.

**Comment by Kyfafyd** [comment]
Thanks for your response!
I would like to learn how to obtain the results in Table 2 (Open-vocabulary LVIS compared to ViLD)?

**Comment by Kyfafyd** [comment]
One more question, it seems the text embeddings of all 1203 classes in LVIS are used in OVOD training, is this normal?
As in OVOD setting, the model can only access base 866 classes.

**Comment by xingyizhou** [comment]
Hi @Kyfafyd ,

  1. Please check [here](https://github.com/facebookresearch/Detic/issues/32) for experiments in Table 2. 
  2. Great question! This is an implementation trick. We used embeddings for all 1203 classes in training, but the other 1203-866 classes are never used in training. Note we used the sigmoid cross-entropy loss so that the classes are independent. We ignored the negative losses for other classes by the flag [here](https://github.com/facebookresearch/Detic/blob/main/configs/BoxSup_ViLD_200e.py#L59).

**Comment by xingyizhou** [comment]
@Kyfafyd what do you mean? Where are 460 epochs from?

**Comment by Kyfafyd** [comment]
@xingyizhou Ah 460 epochs is from the original ViLD paper. Never mind...
However, I have another question, in your ViLD implementation, there is no **knowledge distallation** process right?
